lib/react_on_rails/react_renderer.rb

```diff
@@ -1,20 +1,17 @@
 module ReactOnRails
   class ReactRenderer
 
-    # "this" does not need a closure as it refers to the "this" defined by the
-    # calling the calling context which is the "this" in the execJs environment.
-    def render_js_react_component
-      <<-JS.strip_heredoc
-      function renderReactComponent(componentClass, props) {
-        return this.React.renderToString(
-         componentClass(props)
-        );
-      }
+    # Returns a React element, unwrapping it if the component is a generator function
+    def render_js_react_element(react_component, props)
+      <<-JS
+        #{react_component}.generator ?
+          #{react_component}(#{props}) :
```

Comment:
I wish there was something like this for determining a component (rather than an element)
<img width="688" alt="cursor_and_top-level_api___react" src="https://cloud.githubusercontent.com/assets/1118459/9829248/60b8695a-58b1-11e5-88aa-427c1b55f469.png">

@petehunt We could not figure out a way to distinguish between our own generator function to create a react element, versus the function created by calling React.createClass.

CC: @alexfedoseev